src/LanguageServer/Protocol/Handler/Diagnostics/AbstractWorkspacePullDiagnosticsHandler.cs

```diff
@@ -79,6 +81,11 @@ private void OnLspTextChanged(object? sender, EventArgs e)
         UpdateLspChanged();
     }
 
+    private void OnWorkspaceRefreshRequested()
+    {
+        UpdateLspChanged();
```

Comment:
Can you add a test similar to https://github.com/dotnet/roslyn/blob/main/src/LanguageServer/ProtocolUnitTests/Diagnostics/PullDiagnosticTests.cs#L2019 (instead of making a document edit, it calls the global refresher)?